lib/puppet/defaults.rb

```diff
@@ -32,6 +32,41 @@ def self.valid_file_checksum_types
       %w[sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite md5 md5lite mtime ctime]
   end
 
+  def self.log_ca_migration_warning
+    urge_to_migrate = <<-UTM
+The cadir is currently in your puppet/ssl directory. This location will be
+deprecated and removed in a future version of puppet. Please run the the
+puppetserver ca tool to migrate it out to the puppetserver directory. Use
```

Comment:
I think I don't want to hardcode any paths here, I think we want to keep that specific to the cli tool.

I would like to keep `out` in the message because it really emphasizes that it is leaving the puppet confdir.

Reply:
Why not? Doesn't this PR already hard-code what it expects the new CA dir to be?

If we keep `out`, can we say `out of puppet/ssl and into <blah>`? When reading this, I first thought it said "migrate it out `of` the puppetserver directory", which was confusing.